packages/block-editor/src/components/media-upload/index.native.js

```diff
@@ -136,8 +148,20 @@ export class MediaUpload extends Component {
 	}
 
 	onPickerPresent() {
+		const { autoOpen } = this.props;
+		const isIOS = Platform.OS === 'ios';
+
 		if ( this.picker ) {
-			this.picker.presentPicker();
+			// the delay below is required because on iOS this action sheet gets dismissed by the close event of the Inserter
+			// so this delay allows the Inserter to be closed fully before presenting action sheet.
+			if ( autoOpen && isIOS ) {
+				delay(
+					() => this.picker.presentPicker(),
+					PICKER_OPENING_DELAY
+				);
+			} else {
+				this.picker.presentPicker();
+			}
```

Comment:
Noting that I just created an issue to address looking for an alternative solution for this delay which is used here and in other parts of the codebase when dealing with multiple bottom sheets on iOS: https://github.com/WordPress/gutenberg/issues/29672